src/util/exception_utils.h

```diff
@@ -33,4 +33,23 @@ class invalid_user_input_exceptiont
   std::string what() const noexcept;
 };
 
+class system_exceptiont
```

Comment:
Please add comments: when should this be used, i.e., what is (or isn't) a "system" exception.